content/webapp/routeHandlers/handleNewsletterSignup.js

```diff
@@ -0,0 +1,58 @@
+const fetch = require('isomorphic-unfetch');
+require('dotenv').config();
+const dotmailerCreds = JSON.parse(process.env.dotmailer_creds);
+const { newsletterApiUsername, newsletterApiPassword } = dotmailerCreds;
+
+async function handleNewsletterSignup(ctx, next) {
+  const { addressbookid, email } = ctx.request.body;
+  const newsletterApiUrl = 'https://r1-api.dotmailer.com/v2';
+  const headers = {
+    'Content-Type': 'application/json',
+    Authorization: `Basic ${Buffer.from(
+      `${newsletterApiUsername}:${newsletterApiPassword}`
+    ).toString('base64')}`,
+  };
+  // We first assume the email address is new…
+  const newBody = JSON.stringify({
```

Comment:
Why newbody? Not just body?

Reply:
it's to differentiate it from a 'resubscribe' body. Maybe not the best named variables, though.